cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/ElasticsearchGenerator.scala

```diff
@@ -0,0 +1,430 @@
+package edu.uci.ics.cloudberry.zion.model.impl
+
+import edu.uci.ics.cloudberry.zion.model.datastore.{IQLGenerator, IQLGeneratorFactory, QueryParsingException}
+import edu.uci.ics.cloudberry.zion.model.schema._
+
+import play.api.libs.json._
+
+import scala.collection.mutable
+
+/**
+  * Defines constant strings for query languages supported by Elasticsearch
+  */
+trait ElasticImpl {
+
+  val aggregateFuncMap: Map[AggregateFunc, String]
+
+  def getAggregateStr(aggregate: AggregateFunc): String = {
+    aggregateFuncMap.get(aggregate) match {
+      case Some(impl) => impl
+      case None => throw new QueryParsingException(s"No implementation is provided for aggregate function ${aggregate.name}")
+    }
+  }
+}
+
+object ElasticImpl extends ElasticImpl {
+  val aggregateFuncMap: Map[AggregateFunc, String] = Map(
+    Count -> "count",
+    Max -> "max",
+    Min -> "min",
+    Avg -> "avg",
+    Sum -> "sum"
+  )
+}
+
+
+class ElasticsearchGenerator extends IQLGenerator {
+
+  /**
+    * represent the expression for a [[Field]]
+    *
+    * @param refExpr the expression for referring this field by the subsequent statements
+    * @param defExpr the expression the defines the field
+    *
+    */
+  case class FieldExpr(refExpr: String, defExpr: String)
+
+  /**
+    * Partial parse results after parsing each [[Statement]]
+    *
+    * @param strs    a sequence of parsed query strings, which would be composed together later.
+    * @param exprMap a new field expression map
+    */
+  case class ParsedResult(strs: Seq[String], exprMap: Map[String, FieldExpr])
+
+  protected val allFieldVar: String = "*"
+
+  protected var joinTermsFilter = Seq[Int]() // Global variable: JOIN field
+
+  protected val typeImpl: ElasticImpl = ElasticImpl
+
+  def generate(query: IQuery, schemaMap: Map[String, AbstractSchema]): String = {
+    val temporalSchemaMap = GeneratorUtil.splitSchemaMap(schemaMap)._1
+    val result = query match {
+      case q: Query => parseQuery(q, temporalSchemaMap)
+      case q: CreateView => parseCreate(q, temporalSchemaMap)
+      case q: AppendView => parseAppend(q, temporalSchemaMap)
+      case q: UpsertRecord => parseUpsert(q, schemaMap)
+      case _ => ???
```

Comment:
Since Cloudberry, currently, has no query for DropView and DeleteRecord, we haven't implemented functions for dropping view and deleting record, yet. We can implement it if needed.

Reply:
Now, `DropView` and `DeleteRecord` functions are added to the code.